### DISCUSSION TITLE
RNEA derivatives with respect to inertia parameters

### DISCUSSION THREAD

**Discussion by MaximilianDio** [stack-of-tasks/pinocchio#1887] [Q&A]
Hi there.

I would like to optimize dynamic properties like mass, center of gravity, inertia etc., however, the kinematics and therefore the joint frames do not change. As mentioned in https://github.com/stack-of-tasks/pinocchio/discussions/1874 It is fairly easy to update the model w.r.t. the inertia parameters but is there also a way to get the derivatives of lets say the RNEA w.r.t. to the parameters. I know that the derivatives w.r.t. to the joint configuration, velocities etc are possible, but I have not found anything about inertia parameters. 

kind regards,
Max

**Comment by jcarpent** [comment]
You have the regressors of the dynamic parameters for the system identification. Is what you need?
Otherwise, to get extra derivatives, it is also possible to use autodiff framework like Casadi through Pinocchio to get all the desired derivatives.

Best,
Justin

**Comment by Betancourt20** [comment]
Hello, 

I was looking to the best/optimal way to compute ${ \frac{\partial{({J} {J}^T)}}{\partial{{q}_i}} }$ and I found that using finite difference is quite not the best as explained in https://hal.science/hal-01790971 . Thus, is Casadi through Pinocchio the best way to do it? or do you believe there is another easiest way?

Thank you so much for your support,
best regards,
Julio

**Reply by jcarpent** [reply]
This is another thread of discussion. Could you open a new one please? And I will answer afterwards.
